src/hb-array.hh

```diff
@@ -171,18 +171,21 @@ struct hb_array_t : hb_iter_with_fallback_t<hb_array_t<Type>, Type&>
 
   unsigned int get_size () const { return length * this->get_item_size (); }
 
-  void reverse ()
+  /*
+   * Reverse the order of items in this array in the range [start, end).
+   */
+  void reverse (unsigned start = 0, unsigned end = -1)
```

Comment:
Maybe use `UINT_MAX` or `(unsigned) -1`.  I don't know which one we're using these days.